src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -119,6 +119,312 @@ public static JsonElement CreateJsonElement(string json)
+
+                    return (T)(object)items;
+                }
+                else if (typeof(T) == typeof(int[]))
+                {
+                    int numItems = 0;
+                    // is this an array of properties
+                    foreach (JsonElement j in jsonElement.EnumerateArray())
+                        numItems++;
+
+                    int[] items = new int[numItems];
+                    numItems = 0;
+
+                    foreach (JsonElement j in jsonElement.EnumerateArray())
+                        if (j.TryGetInt32(out int i))
+                            items[numItems++] = i;
+                        else if (int.TryParse(j.GetRawText(), out int value))
+                            items[numItems++] = value;
+                        else
+                            throw LogHelper.LogExceptionMessage(
+                                new JsonException(
+                                    LogHelper.FormatInvariant(
+                                        LogMessages.IDX11028,
+                                        jsonElement.GetRawText(),
+                                        "Integer32")));
+
+                    return (T)(object)items;
+                }
+                else if (typeof(T) == typeof(long[]))
+                {
+                    int numItems = 0;
+                    foreach (JsonElement j in jsonElement.EnumerateArray())
+                        numItems++;
+
+                    long[] items = new long[numItems];
+                    numItems = 0;
+                    foreach (JsonElement j in jsonElement.EnumerateArray())
+                    {
+                        if (j.TryGetInt64(out long l))
+                            items[numItems++] = l;
+                        else if (long.TryParse(j.GetRawText(), out long value))
+                            items[numItems++] = value;
+                        else
+                            throw LogHelper.LogExceptionMessage(
+                                new JsonException(
+                                    LogHelper.FormatInvariant(
+                                        LogMessages.IDX11028,
+                                        jsonElement.GetRawText(),
+                                        "Integer64")));
+                    }
+
+                    return (T)(object)items;
+                }
+            }
+            else if (typeof(T) == typeof(string))
+            {
+                if (jsonElement.ValueKind == JsonValueKind.String)
+                    return (T)(object)jsonElement.GetString();
+
+                return (T)(object)jsonElement.GetRawText();
+            }
+
+            throw LogHelper.LogExceptionMessage(
+                new JsonException(
+                    LogHelper.FormatInvariant(
+                        LogMessages.IDX11027,
+                        typeof(T).ToString(),
+                        jsonElement.ValueKind.ToString())));
```

Comment:
test checking this message is correctly returned?

Reply:
@jennyf19 those should not be thrown if call graph is from TryGet...